### PR TITLE
docs(atomic-react): move atomic-react sample to dedicated package

### PR DESCRIPTION
**Pull request by olamothe**
@jpmarceau / @npushkarskii 

This change will make it so that the link we use here: https://docs.coveo.com/en/atomic/latest/usage/frameworks/atomic-react-wrapper/ ( ie: See this "Atomic React Page" link) will no longer work.

Is it complicated to change it to https://github.com/coveo/ui-kit/blob/master/packages/samples/atomic-react/src/App.tsx when this gets merged ? (if you would like to see it before the PR gets merged, it's available at: https://github.com/coveo/ui-kit/blob/KIT-1560/packages/samples/atomic-react/src/App.tsx )

https://coveord.atlassian.net/browse/KIT-1560

### CODE REVIEW

File: packages/samples/atomic-react/public/app.css [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+  --spacing: 16px;
+  --spacing-col: 32px;
+  --search-box-width: 648px;
+}
+
+atomic-search-box {
+  margin-top: 1rem;
+}
+
+atomic-search-interface {
```
Comment (ThibodeauJF) [inline packages/samples/atomic-react/public/app.css:51]: What's the support status of layouts for Atomic React ?

Comment (olamothe) [reply packages/samples/atomic-react/public/app.css:51]: Should work... I could do the change in this PR 👍

**Comment by liviarett** [comment]
@olamothe I don't know if I'm being super dumb here, but I am on your branch trying to test your react page with the folded component, and..... the changes I make don't reflect on the page! the webpack console pretends it's done its job, but it doesn't seem to change anything in `public/dist`. any steps I might have missed?

**Comment by liviarett** [comment]
I think my comment above is related to the reason why the test is failing

**Comment by olamothe** [comment]
> @olamothe I don't know if I'm being super dumb here, but I am on your branch trying to test your react page with the folded component, and..... the changes I make don't reflect on the page! the webpack console pretends it's done its job, but it doesn't seem to change anything in `public/dist`. any steps I might have missed?

Yep, was missing a build step in the `dev` command

**Pull request merged**